File: Dijkstra-Algorithm.py
# Dijkstra's Algorithm finds the shortest paths from a start node to all other nodes in a weighted graph.

import heapq

def dijkstra(graph, start):
    distances = {vertex: float('infinity') for vertex in graph}
    distances[start] = 0
    pq = [(0, start)]

    while pq:
        current_distance, current_vertex = heapq.heappop(pq)

        for neighbor, weight in graph[current_vertex].items():
            distance = current_distance + weight

            if distance < distances[neighbor]:
                distances[neighbor] = distance
                heapq.heappush(pq, (distance, neighbor))

    return distances

# Example usage assumes a graph represented as a dictionary of dictionaries.

File: Dijkstra-Algorithm.js
// Dijkstra's Algorithm finds the shortest paths from a start node to all other nodes in a weighted graph.

function dijkstra(graph, start) {
    const distances = {};
    const prev = {};
    const visited = new Set();
    Object.keys(graph).forEach(node => distances[node] = Infinity);
    distances[start] = 0;

    const queue = new PriorityQueue();
    queue.enqueue(start, 0);

    while (!queue.isEmpty()) {
        const {element: node} = queue.dequeue();
        visited.add(node);

        Object.keys(graph[node]).forEach(neighbor => {
            if (!visited.has(neighbor)) {
                const alt = distances[node] + graph[node][neighbor];
                if (alt < distances[neighbor]) {
                    distances[neighbor] = alt;
                    prev[neighbor] = node;
                    queue.enqueue(neighbor, distances[neighbor]);
                }
            }
        });
    }

    return distances;
}

// Note: PriorityQueue class implementation is required here. This example assumes its availability.
